lib/spack/spack/spec.py

```diff
@@ -1015,6 +1034,11 @@ class SpecBuildInterface(lang.ObjectWrapper):
         default_handler=_libs_default_handler
     )
 
+    cmake_args = ForwardQueryToPackage(
+        'cmake_args',
+        default_handler=_cmake_args_default_handler
+    )
```

Comment:
Sorry to come this late, I think I missed this PR when it was first submitted. That said I don't think we need a mechanism like `ForwardQueryToPackage` for something like this.  It would be probably sufficient to agree on a method to be implemented by all lapack/blas providers.

Reply:
I modeled this after how `blas_libs` and `lapack_libs` are implemented. That is, packages that `depend_on('blas')` can use `self.spec['blas'].libs` to get the correct property. The idea is that CMake packages that use the standard `FindBLAS` (or `FindLAPACK`) module can use `self.spec['blas'].cmake_args` in their `cmake_args()` method.

There's probably a better way of doing this—`FindBLAS` and `FindLAPACK` are hard to control and don't support all BLAS/LAPACK libraries well—but I think this is an improvement over "just put the prefixes in `CMAKE_PREFIX_PATH` and hope the correct package is picked up by the find module".